### PR TITLE
Sets up tooling to create single executable UI package

### PR DESCRIPTION
**Pull request by vsreekanti**
This PR adds support for using the [pkg](https://github.com/vercel/pkg) library to generate a single executable for the UI. In order to do that, we add our own custom entrypoint (`server.js`) which is a very thin wrapper around the default `next` server. 

To generate the package, we can run `pkg --compress -t node16-linux-x64,node16-macos-x64,node16-macos-arm64 . --out-path pkg`, which will generate the 3 executables we care about in the `pkg` directory. 

## Tests
Works on an Ubuntu machine. Still needs to be validated in other environments.

### CODE REVIEW

File: src/ui/common/src/components/workflows/ReactFlowCanvas.tsx [hunk 1]
```diff
@@ -42,7 +42,7 @@ const ReactFlowCanvas: React.FC<ReactFlowCanvasProps> = ({
     setTimeout(fitView, 200);
   }, [openSideSheetState]);
 
-  useEffect(fitView);
+  useEffect(fitView, []);
```
Comment (agiron123) [inline src/ui/common/src/components/workflows/ReactFlowCanvas.tsx:45]: Should be resolved in https://github.com/aqueducthq/aqueduct/pull/23 can you update here to match?

File: src/ui/app/server.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+require('dotenv').config()
```
Comment (agiron123) [inline src/ui/app/server.js:1]: Any changes here that needed to happen to properly handle cookies? Or was I just hitting issues with env variables not being properly set?

**Comment by agiron123** [comment]
Could you update our makefile to add a package action, and add that to a .github action as well please?

**Review by agiron123 — APPROVED** [review]
Looks good! Excited to see this working.

Will pull down shortly to test out on my M1 mac.

File: src/ui/app/server.js (hunk 2 above)
Comment (vsreekanti) [reply src/ui/app/server.js:1]: I think it was the env variables issue. Once I fixed this, everything worked as expected for me.

**Pull request closed without merging**